contracts/minter-suite/Minters/MinterHolder/MinterHolderV2.sol

```diff
@@ -0,0 +1,795 @@
+ * ----------------------------------------------------------------------------
+ * This contract allows gated minting with support for vaults to delegate minting
+ * privileges via an external delegation registry. This means a vault holding an
+ * allowed token can delegate minting privileges to a wallet that is not holding an
+ * allowed token, enabling the vault to remain air-gapped while still allowing minting.
+ * The delegation registry contract is responsible for managing these delegations,
+ * and is available at the address returned by the public immutable
+ * `delegationRegistryAddress`. At the time of writing, the delegation
+ * registry enables easy delegation configuring at https://delegate.cash/.
+ * Art Blocks does not guarentee the security of the delegation registry, and
+ * users should take care to ensure that the delegation registry is secure.
+ * Delegations must be configured by the vault owner prior to purchase. Supported
+ * delegation types include token-level, contract-level (via genArt721CoreAddress), or
+ * wallet-level delegation. Contract-level delegations must be configured for the core
+ * token contract as returned by the public immutable variable `genArt721CoreAddress`.
+ */
+contract MinterHolderV2 is ReentrancyGuard, IFilteredMinterHolderV1 {
+    /// Delegation registry address
+    address public immutable delegationRegistryAddress;
+
+    /// Delegation registry address
+    IDelegationRegistry private immutable delegationRegistryContract;
+
+    /**
+     * @notice Registered holders of NFTs at address `_NFTAddress` to be
+     * considered for minting.
+     */
+    event RegisteredNFTAddress(address indexed _NFTAddress);
+
+    /**
+     * @notice Unregistered holders of NFTs at address `_NFTAddress` to be
+     * considered for minting.
+     */
+    event UnregisteredNFTAddress(address indexed _NFTAddress);
+
+    /**
+     * @notice Allow holders of NFTs at addresses `_ownedNFTAddresses`, project
+     * IDs `_ownedNFTProjectIds` to mint on project `_projectId`.
+     * `_ownedNFTAddresses` assumed to be aligned with `_ownedNFTProjectIds`.
+     * e.g. Allows holders of project `_ownedNFTProjectIds[0]` on token
+     * contract `_ownedNFTAddresses[0]` to mint.
+     */
+    event AllowedHoldersOfProjects(
+        uint256 indexed _projectId,
+        address[] _ownedNFTAddresses,
+        uint256[] _ownedNFTProjectIds
+    );
+
+    /**
+     * @notice Remove holders of NFTs at addresses `_ownedNFTAddresses`,
+     * project IDs `_ownedNFTProjectIds` to mint on project `_projectId`.
+     * `_ownedNFTAddresses` assumed to be aligned with `_ownedNFTProjectIds`.
+     * e.g. Removes holders of project `_ownedNFTProjectIds[0]` on token
+     * contract `_ownedNFTAddresses[0]` from mint allowlist.
+     */
+    event RemovedHoldersOfProjects(
+        uint256 indexed _projectId,
+        address[] _ownedNFTAddresses,
+        uint256[] _ownedNFTProjectIds
+    );
```

Comment:
Makes sense! Yeah sounds like it's the perfect time to make this change.

With these moved, the ordering of the contract's registry assignments will look like this (after the other nits to re-arrange these after the events):
```
contract MinterHolderV2 is ReentrancyGuard, IFilteredMinterHolderV1 {
    // add Enumerable Set methods
    using EnumerableSet for EnumerableSet.AddressSet;

    /// Delegation registry address
    address public immutable delegationRegistryAddress;

    /// Delegation registry address
    IDelegationRegistry private immutable delegationRegistryContract;
```

Reply:
this is perfect! that `using x for y` is great to have at the very top of the contract IMO. I can't find anywhere in the solidity style guide stating where it should be, but I much prefer it at the top since it defines the library functions available across the entire contract 😄